glean-core/ffi/src/handlemap_ext.rs

```diff
@@ -0,0 +1,114 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Glean does not pass errors through the FFI component upwards.
+//! Most errors are not propagated and simply only logged for debugging.
+//! Errors should be recoverable and the platform-side should mostly ignore them.
+//! Additionally the platform-side can work asynchronously anyway, leaving us with no good way to
+//! pass back errors.
+//!
+//! The `HandleMapExtension` extension traits adds additional methods that log potential errors.
+//!
+//! **Note: the platform-side still needs to check for null pointers or other default values before
+//! using returned values. This is only relevant for creation of the main object and metrics.
```

Comment:
nit: is it worth mentioning that this is true because the handles of these objects are stored in `ConcurrentHandleMap`(s)?

Reply:
It's more because of that's nearly the only part where we actually return something potentially fallible.

Reply:
I see, please call that out in the comment there, so that it's clearer (and so that next time I can avoid asking you the same question again! :D )

Reply:
Will do!